pkgs/development/python-modules/rapidocr-onnxruntime/default.nix

```
@@ -0,0 +1,134 @@
+    "PP-OCRv1/ch_ppocr_mobile_v2.0_cls_infer.onnx" = "sha256-5HrO32YyMPiGP/GrDmTdLYK4OPzrWVcUbasYWonWIVw=";
+  };
+
+  models = lib.pipe modelFiles [
+    (lib.mapAttrsToList (
+      name: hash: {
+        inherit name;
+        path = fetchurl {
+          url = "https://huggingface.co/SWHL/RapidOCR/resolve/${modelsRev}/${name}";
+          inherit hash;
+        };
+      }
+    ))
+    (linkFarm "rapidocr-onnxruntime-models")
+  ];
+in
+buildPythonPackage {
+  pname = "rapidocr-onnxruntime";
+  inherit version src;
+  pyproject = true;
+
+  sourceRoot = "${src.name}/python";
+
+  # The setup.py supplied by upstream tries to determine the current version by
+  # fetching the latest version of the package from PyPI, and then bumping the version number.
+  # This is not allowed in the Nix build environment as we do not have internet access,
+  # hence we patch that out and get the version from the build environment directly.
+  patches = [ ./setup-py-override-version-checking.patch ];
+
+  nativeBuildInputs = [ pythonRelaxDepsHook ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    pyclipper
+    opencv4
+    numpy
+    six
+    shapely
+    pyyaml
+    pillow
+    onnxruntime
+  ];
+
+  # Don't use vendored OpenCV
+  pythonRemoveDeps = [ "opencv-python" ];
+
+  pythonImportsCheck = [ "rapidocr_onnxruntime" ];
+
+  # These are tests for different backends.
+  # Why are there tests for different backends in the same test folder? Good question!
+  disabledTestPaths = [
+    "tests/test_vino.py"
+    "tests/test_paddle.py"
+  ];
+
+  # HACK:
+  # Upstream uses a very unconventional structure to organize the packages, and we have to coax the
+  # existing infrastructure to work with it.
+  # See https://github.com/RapidAI/RapidOCR/blob/02829ef986bc2a5c4f33e9c45c9267bcf2d07a1d/.github/workflows/gen_whl_to_pypi_rapidocr_ort.yml#L83-L85
+  # for the "intended" way of building this package.
+
+  postPatch = ''
+    mv setup_onnxruntime.py setup.py
+
+    mkdir -p rapidocr_onnxruntime/models
+    # Flatten the structure so that all models are placed in the same directory
+    find ${models} -name "*.onnx" -exec ln -st rapidocr_onnxruntime/models {} +
+
+    # Magic patch from upstream - what does this even do??
+    echo "from .rapidocr_onnxruntime.main import RapidOCR, VisRes" > __init__.py
+  '';
+
+  # Upstream expects the source files to be under rapidocr_onnxruntime/rapidocr_onnxruntime
+  # instead of rapidocr_onnxruntime for the wheel to build correctly.
+  preBuild = ''
+    mkdir rapidocr_onnxruntime_t
+    mv rapidocr_onnxruntime rapidocr_onnxruntime_t
+    mv rapidocr_onnxruntime_t rapidocr_onnxruntime
+  '';
+
+  # Revert the above hack
+  postBuild = ''
+    mv rapidocr_onnxruntime rapidocr_onnxruntime_t
+    mv rapidocr_onnxruntime_t/* .
+  '';
+
+  meta = {
+    description = "Cross platform OCR Library based on OnnxRuntime";
+    homepage = "https://github.com/RapidAI/RapidOCR";
```

```suggestion
    homepage = "https://github.com/RapidAI/RapidOCR";
    changelog = "https://github.com/RapidAI/RapidOCR/releases/tag/v${version}";
```
Adding meta.changelog is useful for future updates.

---

Fixed.